src/runtime_src/core/common/xclbin_parser.cpp

```diff
@@ -393,13 +393,19 @@ get_softkernels(const axlf* top)
       auto begin = reinterpret_cast<const char*>(top) + pSection->m_sectionOffset;
       auto soft = reinterpret_cast<const soft_kernel*>(begin);
 
-      softkernel_object sko;
+      sks.emplace_back(softkernel_object{});
+      auto sko = sks.back();
+      //softkernel_object sko;
       sko.ninst = soft->m_num_instances;
-      sko.symbol_name = const_cast<char*>(begin + soft->mpo_symbol_name);
+      
+      //sko.symbol_name = const_cast<char*>(begin + soft->mpo_symbol_name);
+      sko.symbol_name = std::string(begin + soft->mpo_symbol_name);
+      sko.mpo_name = std::string(begin + soft->mpo_name);
+      sko.mpo_version = std::string(begin + soft->mpo_version);
```

Comment:
This sort of copying worries me, what guarantee is there that mpo_symbol_name, mpo_name, mpo_version are null terminated?   Yes, same problem was there before, but can we fix this?

Reply:
there is no size limit on these soft kernel strings. What do you suggest?

Reply:
The code is wrong then (unless I am missing something).   I just looked at xclbin.h, all the _name vars are numbers (uint32_t).  You can't construct a std::string from this using std::string(begin + number) because this string constructor interprets 'begin + number' as the address of a null terminated char* string, which it clearly is not.   The old code was okay, because it just reinterpreted the number as a char*.

Has these code changes been tested?